helpers/setup-spark-k8s.py

```diff
@@ -0,0 +1,135 @@
+    print('Looks like either kubernetes is not setup properly or default kubeconfig file is not accessible!')
+    print('Please take the following remedial actions.')
+    print('	1. Please setup kubernetes and make sure kubeconfig file is available, accessible and correct.')
+    print('	2. sudo snap connect spark-client:enable-kubeconfig-access')
+
+def extract_default_cluster_from_kubeconfig(kubeconfig: str) -> str:
+    with open(kubeconfig) as f:
+        kube_cfg = yaml.safe_load(f)
+        contexts = [n['name'] for n in kube_cfg['contexts']]
+        clusters = [n['context']['cluster'] for n in kube_cfg['contexts']]
+        try:
+            context_id = contexts.index( kube_cfg['current-context'])
+        except ValueError:
+            print('Default context invalid or not set in kubeconfig file, please try again.')
+            sys.exit(-200)
+        return clusters[context_id]
+
+def try_extract_default_cluster_from_kubeconfig(kubeconfig: str) -> str:
+    try:
+        cluster_name = extract_default_cluster_from_kubeconfig(kubeconfig)
+    except IOError as e:
+        print_help_for_missing_or_inaccessible_kubeconfig_file()
+        sys.exit(-300)
+    return cluster_name
+
+def extract_ca_crt_from_kube_config(kubeconfig: str, cluster_name: str = None) -> None:
+    with open(kubeconfig) as f:
+        kube_cfg = yaml.safe_load(f)
+        cluster_names = [n['name'] for n in kube_cfg['clusters']]
+        certs = [n['cluster']['certificate-authority-data'] for n in kube_cfg['clusters']]
+
+        if cluster_name:
+            try:
+                cluster_id = cluster_names.index(cluster_name)
+            except ValueError:
+                print('Invalid cluster selection, please try again....')
+                sys.exit(-100)
+        elif len(cluster_names) > 1:
+            print ('\n'.join(["{}. {}".format(cluster_names.index(a), a) for a in cluster_names]))
+            print('\nPlease select a cluster:')
+            cluster_id = input()
+
+            try:
+               int(cluster_id)
+            except ValueError:
+                print('Invalid cluster selection, please try again....')
+                sys.exit(-100)
+
+            if int(cluster_id) not in range(len(cluster_names)):
+                print('Invalid cluster selection, please try again....')
+                sys.exit(-100)
+        elif len(cluster_names) == 1:
+            cluster_id = 0
+        else:
+            print('ERROR: No clusters found in kubeconfig file. Please provide a valid kubeconfig file!')
+            sys.exit(-100)
+
+        os.system('\n\necho {} | base64 --decode'.format(certs[int(cluster_id)]))
+
+
+def setup_user(username: str, namespace: str, kubeconfig: str, cluster_name: str) -> None:
+    clusterrolebindingname = username + '-role'
+    clusterroleaccess='view'
+    os.system("{} create serviceaccount --kubeconfig={} --cluster={} {} --namespace={}".format(kubectl_cmd, kubeconfig, cluster_name, username, namespace))
```

Comment:
I think we should not work with `--cluster` at all. We should work with `--context` instead.
Context is better because it also has the user information.